config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/environment/CharacterBlacklist.java

```diff
@@ -0,0 +1,52 @@
+package com.ca.apim.gateway.cagatewayconfig.util.environment;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * Contains all blacklisted characters for file names in Windows and UNIX. Purpose of class is to provide all
+ * values in a set.
+ */
+public class CharacterBlacklist {
+
+    // Windows
+    public static final char LESS_THAN = '<';
+    public static final char GREATER_THAN = '>';
+    public static final char COLON = ':';
+    public static final char DOUBLE_QUOTE = '\'';
+    public static final char FORWARD_SLASH = '/';
+    public static final char BACK_SLASH = '\\';
+    public static final char PIPE = '|';
+    public static final char QUESTION_MARK = '?';
+    public static final char ASTERISK = '*';
+
+    // Unix
+    public static final char NULL_CHAR = '\0';
+
+    private static final Field[] fields = CharacterBlacklist.class.getFields();
+
+    /**
+     * Retrieve set of blacklist characters
+     *
+     * @return Character Set with all constants in the class. Constants are blacklisted characters.
+     */
+    public static Set<Character> getCharBlacklist() {
+        CharacterBlacklist cblInstance = new CharacterBlacklist();
```

Comment:
you can create a static final field to initialize the charBlackList, and run this method only once instead of only having the field list as static. 
And for getting the field value you don't need an instance, as the fields are static, you can do field.get(CharacterBlacklist.class)